src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseTestCase.java

```diff
@@ -231,4 +233,29 @@ public String toString() {
         return "ProgrammingExerciseTestCase{" + "id=" + getId() + ", testName='" + testName + '\'' + ", weight=" + weight + ", active=" + active + ", visibility=" + visibility
                 + ", bonusMultiplier=" + bonusMultiplier + ", bonusPoints=" + bonusPoints + '}';
     }
+
+    /**
+     * Check if the provided test was found in the result's feedbacks with positive = true.
+     *
+     * @param result of the build run.
+     * @return true if there is a positive feedback for a given test.
+     */
+    public boolean isSuccessful(Result result) {
```

Comment:
Should this and the next method be part of the test case? Or would it be better placed in the result and pass the test case as argument there?
My reasoning would be that the feedbacks belong to the result and then we want to check if one of those feedbacks matches a given test case.

Reply:
I was more pragmatic in moving it into test case as test case class is currently a lot smaller than the result class

Reply:
Also we want to check if a test case is successful so I think having it in test case is fine here